supervisor/supervisor.go

```diff
@@ -0,0 +1,124 @@
+package supervisor
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"github.com/oklog/ulid/v2"
+	"github.com/open-telemetry/opamp-go/client"
+	"github.com/open-telemetry/opamp-go/supervisor/agent"
+	"log"
+	"math/rand"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+)
+
+type Logger struct {
+	Logger *log.Logger
+}
+
+func (l *Logger) Debugf(format string, v ...interface{}) {
+	l.Logger.Printf(format, v...)
+}
+
+func (l *Logger) Errorf(format string, v ...interface{}) {
+	l.Logger.Printf(format, v...)
+}
+
+type supervisor struct {
+	logger *Logger
+
+	configFile string
+	config     *Configuration
```

Comment:
Should we also externalize the `Configuration` as an interface that we load externally and pass to the `supervisor`? If we do it then we can have specialize implementations of the `Configuration` that have a different yaml format and for example no longer require `args` section because knowledge about command line arguments is hard-coded in that specialized implementation.